Sources/TuistHasher/ResourcesContentHasher.swift

```diff
@@ -3,43 +3,78 @@ import TuistCore
         self.init(
             contentHasher: contentHasher,
-            privacyManifestContentHasher: PrivacyManifestContentHasher(contentHasher: contentHasher)
+            privacyManifestContentHasher: PrivacyManifestContentHasher(contentHasher: contentHasher),
+            platformConditionContentHasher: PlatformConditionContentHasher(contentHasher: contentHasher)
         )
     }
 
     public init(
         contentHasher: ContentHashing,
-        privacyManifestContentHasher: PrivacyManifestContentHasher
+        privacyManifestContentHasher: PrivacyManifestContentHasher,
+        platformConditionContentHasher: PlatformConditionContentHashing
     ) {
         self.contentHasher = contentHasher
         self.privacyManifestContentHasher = privacyManifestContentHasher
+        self.platformConditionContentHasher = platformConditionContentHasher
     }
 
     // MARK: - ResourcesContentHashing
 
-    public func hash(resources: ResourceFileElements) throws -> String {
-        var hashes = try resources.resources
-            .sorted(by: { $0.path < $1.path })
-            .map { try contentHasher.hash(path: $0.path) }
+    public func hash(identifier: String, resources: ResourceFileElements) throws -> MerkleNode {
+        var children: [MerkleNode] = []
+
+        children
+            .append(
+                contentsOf: try resources.resources.sorted(by: { $0.path < $1.path })
+                    .map { try hashResourceFileElement(element: $0) }
+            )
 
         if let privacyManifest = resources.privacyManifest {
-            hashes.append(try privacyManifestContentHasher.hash(privacyManifest))
+            children.append(try privacyManifestContentHasher.hash(
+                identifier: "privacyManifest",
+                privacyManifest: privacyManifest
+            ))
+        }
+
+        return MerkleNode(
+            hash: try contentHasher.hash(children.map(\.hash)),
+            identifier: identifier,
+            children: children
+        )
+    }
+
+    private func hashResourceFileElement(element: ResourceFileElement) throws -> MerkleNode {
+        var children: [MerkleNode] = []
+
+        children.append(MerkleNode(hash: try contentHasher.hash(path: element.path), identifier: "content"))
+        children.append(MerkleNode(hash: try contentHasher.hash(element.isReference), identifier: "isReference"))
+        children.append(MerkleNode(hash: try contentHasher.hash(element.tags), identifier: "tags"))
+        if let inclusionCondition = element.inclusionCondition {
+            children.append(try platformConditionContentHasher.hash(
+                identifier: "inclusionCondition",
+                platformCondition: inclusionCondition
+            ))
         }
 
-        return try contentHasher.hash(hashes)
+        return MerkleNode(
+            hash: try contentHasher.hash(path: element.path),
+            identifier: element.path.pathString,
+            children: []
```

Comment:
```suggestion
            children: children
```

Shouldn't this include the `children`? It seems like we're not using them at all if I'm reading this function right